### ISSUE TITLE
Remove `BuildNameToCertificate()`?

### ISSUE THREAD

**Issue by ctlong** [cloudfoundry/tlsconfig#28]
tldr; can we safely remove use of `BuildNameToCertificate()` in our code?

`WithIdentity()` currently calls `BuildNameToCertificate()`, which has been deprecated since Go 1.14.

https://github.com/cloudfoundry/tlsconfig/blob/3cc0b1c5221ecab87b48220f51c4fb76b124ae4a/config.go#L143-L144

We could remove this method call, which would leave the `NameToCertificate` field set to `nil`, and "allow the library to select the first compatible chain from Certificates." Doing so would change [some under the hood functionality](https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/crypto/tls/common.go;l=1094-1116;drc=178080740c1bc33f2c7f164504eedc24210bbf1e) though, and I'm not confident whether that's okay or not.

The git history of this repo does not appear very helpful, it seems this function call was added prior to its deprecation: see https://github.com/cloudfoundry/tlsconfig/issues/8. And I can't find any other mention of this field to determine whether keeping this call is a feature or a bug.

**Comment by geofffranks** [comment]
Fixed via https://github.com/cloudfoundry/tlsconfig/pull/41
